src/som/interpreter/nodes/ExceptionSignalingNode.java

```diff
@@ -0,0 +1,51 @@
+package som.interpreter.nodes;
+
+import com.oracle.truffle.api.nodes.Node;
+import com.oracle.truffle.api.source.SourceSection;
+
+import bd.primitives.nodes.PreevaluatedExpression;
+import som.vm.Symbols;
+import som.vm.constants.KernelObj;
+import som.vmobjects.SClass;
+import som.vmobjects.SObject;
+
+
+public class ExceptionSignalingNode extends Node {
+
+  @Child protected ExpressionNode getExceptionClassNode;
+  @Child protected ExpressionNode signalExceptionNode;
+  protected final SObject         module;
+
+  public ExceptionSignalingNode(final String exceptionClassName,
+      final String signallingMethodName, final SourceSection sourceSection) {
+    this(exceptionClassName, signallingMethodName, sourceSection, KernelObj.kernel);
+  }
+
+  public ExceptionSignalingNode(final String exceptionClassName,
+      final String signallingMethodName, final SourceSection sourceSection,
+      final SObject theModule) {
+    super();
+    getExceptionClassNode = MessageSendNode.createGeneric(
+        Symbols.symbolFor(exceptionClassName), null,
+        sourceSection);
+    signalExceptionNode = MessageSendNode.createGeneric(
+        Symbols.symbolFor(signallingMethodName),
+        new ExpressionNode[] {getExceptionClassNode},
+        sourceSection);
+    module = theModule;
+  }
+
+  public Object execute(final Object... args) {
+    SClass exceptionClass =
+        (SClass) ((PreevaluatedExpression) getExceptionClassNode).doPreEvaluated(null,
+            new Object[] {module});
+
+    Object[] allArgs = new Object[args.length + 1];
+    allArgs[0] = exceptionClass;
+    for (int i = 0; i < args.length; i++) {
+      allArgs[i + 1] = args[i];
+    }
```

Comment:
Would be nice to extract line 43-47 into a named method

Reply:
Done